src/docker/engine.rs

```diff
@@ -32,6 +33,29 @@ impl EngineType {
     pub fn is_docker(&self) -> bool {
         matches!(self, Self::Docker)
     }
+
+    /// Returns `true` if the engine build supports the `--output` flag.
```

Comment:
I think the wording here is a bit awkward, maybe
```suggestion
    /// Returns `true` if the build command supports the `--output` flag.
```

Reply:
Same for the other fns